File: 206. Reverse Linked List/206.solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function (head) {
  let temp = head;
  let prevVal = null;

  while (temp !== null) {
    const nextTemp = temp.next;
    temp.next = prevVal;
    prevVal = temp;
    temp = nextTemp;
  }

  return prevVal;
};

File: 206. Reverse Linked List/206.solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
  public ListNode reverseList(ListNode head) {
    ListNode temp = head;
    ListNode prev = null;
    
    while (temp != null) {
      ListNode nextTemp = temp.next;
      temp.next = prev;
      prev = temp;
      temp = nextTemp;
    }
    
    return prev;
  }
}